From 5132602916e3d2918a264f950ca6002a8be3af15 Mon Sep 17 00:00:00 2001
From: Szehon Ho <sz.ho@criteo.com>
Date: Wed, 27 Jul 2016 10:10:41 -0700
Subject: [PATCH 0793/1164] CDH-46225 HIVE-14313 : Test failure
 TestMetaStoreMetrics.testConnections (Naveen
 Gangam via Szehon)

(cherry picked from commit df3c5d05968f9abf4f2b101b9dc208c205f2253b)

Change-Id: Ie0bb0a89d2e2ef580d6a1f6a60135862396978bb
---
 .../hive/common/metrics/MetricsTestUtils.java      |    2 +-
 .../hive/metastore/TestMetaStoreMetrics.java       |    9 +++++----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/common/src/test/org/apache/hadoop/hive/common/metrics/MetricsTestUtils.java b/common/src/test/org/apache/hadoop/hive/common/metrics/MetricsTestUtils.java
index c90a614..892e17c 100644
--- a/common/src/test/org/apache/hadoop/hive/common/metrics/MetricsTestUtils.java
+++ b/common/src/test/org/apache/hadoop/hive/common/metrics/MetricsTestUtils.java
@@ -50,7 +50,7 @@ public static void verifyMetricsJson(String json, MetricsCategory category, Stri
     Assert.assertEquals(expectedValue.toString(), jsonNode.asText());
   }
 
-  private static JsonNode getJsonNode(String json, MetricsCategory category, String metricsName) throws Exception {
+  public static JsonNode getJsonNode(String json, MetricsCategory category, String metricsName) throws Exception {
     ObjectMapper objectMapper = new ObjectMapper();
     JsonNode rootNode = objectMapper.readTree(json);
     JsonNode categoryNode = rootNode.path(category.category);
diff --git a/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java b/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java
index 98cb3ec..3f08b88 100644
--- a/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java
+++ b/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java
@@ -140,23 +140,24 @@ public void testConnections() throws Exception {
 
     //initial state is one connection
     String json = metrics.dumpJson();
-    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.OPEN_CONNECTIONS, 1);
+    int initialCount = (new Integer((MetricsTestUtils.getJsonNode(json, MetricsTestUtils.COUNTER,
+                                       MetricsConstant.OPEN_CONNECTIONS)).asText())).intValue();
 
     //create two connections
     HiveMetaStoreClient msc = new HiveMetaStoreClient(hiveConf);
     HiveMetaStoreClient msc2 = new HiveMetaStoreClient(hiveConf);
 
     json = metrics.dumpJson();
-    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.OPEN_CONNECTIONS, 3);
+    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.OPEN_CONNECTIONS, initialCount + 2);
 
     //close one connection, verify still two left
     msc.close();
     json = metrics.dumpJson();
-    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.OPEN_CONNECTIONS, 2);
+    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.OPEN_CONNECTIONS, initialCount + 1);
 
     //close one connection, verify still one left
     msc2.close();
     json = metrics.dumpJson();
-    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.OPEN_CONNECTIONS, 1);
+    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.OPEN_CONNECTIONS, initialCount);
   }
 }
-- 
1.7.9.5

